R/format_metrics_data.R

```diff
@@ -31,5 +33,43 @@ format_metrics_current_data <- function(x) {
 #'   Formats metrics data passed by range query.
 #' @rdname utilities
 format_metrics_range_data <- function(x) {
-  return(x)
+  checkmate::assert_data_frame(x = x,
+                               min.rows = 1,
+                               min.cols = 2)
+  within(data = setNames(
+    object = x$metric,
+    nm = c("metric", "instance", "job")
+  ),
+  expr = {
+    port = as.integer(gsub(
+      pattern = "(.*):(.*)",
+      replacement = "\\2",
+      x = instance
+    ))
+
+    instance = gsub(pattern = "(.*):(.*)",
+                    replacement = "\\1",
+                    x = instance)
+  }) -> x_metrics
+
+  lapply(
+    X = x$values,
+    FUN = function(value_pair) {
+      setNames(object = as.data.frame(value_pair),
+               nm = c("timestamp", "value")) -> df_ts_val
+    }
+  ) -> dfs_to_bind
+
+  i <- 1
+  Reduce(rbind,
+         lapply(
+           X = dfs_to_bind,
+           FUN = function(dfs) {
+             suppressWarnings(cbind(x_metrics[i,], dfs)) -> x
+             i <<- i + 1
```

Comment:
@glenn-m This feels weird and I would change it at some point. At the moment, didn't have ideas on how to do it better without falling on additional packages and didn't want to inflate dependencies only for that.

Reply:
I'm in the process of reading through this, but does `bind(x_metrics[i,], dfs, row.names = NULL)` not resolve this?